### PR TITLE
MCLOUD-5671: Add cache clearing to CLI

### PR DESCRIPTION
**Pull request by mohanelamurugan**
[MCLOUD-5671](https://jira.corp.magento.com/browse/MCLOUD-5671): Add cache clearing to CLI

### Description

For repetitive tasks during development. It is often necessary to clear cache from outside the Magento application, So I have added a CLI method to clear redis cache

### Fixed Issues (if relevant)

1. [MCLOUD-5671](https://jira.corp.magento.com/browse/MCLOUD-5671): magento/magento-cloud-docker#179: Add cache clearing to CLI

### Manual testing scenarios

1. Run docker-compose run deploy redis-clean to clean redis cache outside of Magento application

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages

releasenotes
Added two `bin/magento-docker` CLI commands to clear the Magento cache during development
`flush-redis clears redis cache`
`flush-varnish clears varnish cache`

### CODE REVIEW

File: images/php/cli/bin/redis-clean [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -e
+
+[ "$DEBUG" = "true" ] && set -x
+
+echo "Clearing Redis cache.."
+exec redis redis-cli FLUSHALL
```
Comment (bbatsche) [inline images/php/cli/bin/redis-clean]: I think there may be some confusion here. `exec redis...` leads me to believe you're trying to run `redis-cli` in the `redis` container, however in a bash script such as this the `exec` statement completely different from `docker exec`.

In addition, `redis-cli` needs to know the host it should be communicating with (even when running from within the `redis` container). So you will need to add `-h redis` to the `redis-cli` call.

**Review by bbatsche — CHANGES_REQUESTED** [review]
I'm not sure about the utility of this script or the logic of putting it in the PHP images. Ideally it would be in the `redis` image, but we are using essentially the stock image from Docker.

Ultimately though, this same functionality could be accomplished by running

```bash
docker-compose exec redis redis-cli -h redis FLUSHALL
```

so I'm not sure this solution is much better.

**Comment by michaelcasey316** [comment]
Putting helper commands in the `bin/` folder would be new and different. We already have helper commands in `dist/bin/magento-docker`. Perhaps it would be better to move flush-redis and flush-varnish to `dist/bin/magento-docker` with the rest of the helper commands. What do you all think?

**Comment by aepod** [comment]
If this approach is taken, we also need to modify composer.json to include
these into bin/ directory when composer install is run.

I think this is a better approach.

**Comment by michaelcasey316** [comment]
> If this approach is taken, we also need to modify composer.json to include these into bin/ directory when composer install is run. I think this is a better approach.

We actually already do this mapping: https://github.com/magento/magento-cloud-docker/blob/4ca9932f51234d9359f78a6f5676fd91aca40427/composer.json#L87-L90

Edit: Oh, if you mean including them as extra files alongside `dist/bin/magento-docker`, then yes we need to add the mapping for them as well. My proposal was that they be added as extra commands within `dist/bin/magento-docker`, but separate files is valid idea too.

**Comment by aepod** [comment]
Exactly, great catch. Mohan will update.

**Comment by mohanelamurugan** [comment]
> I'm not sure about the utility of this script or the logic of putting it in the PHP images. Ideally it would be in the `redis` image, but we are using essentially the stock image from Docker.
> 
> Ultimately though, this same functionality could be accomplished by running
> 
> ```shell
> docker-compose exec redis redis-cli -h redis FLUSHALL
> ```
> 
> so I'm not sure this solution is much better.

This approach has been changed now.

**Comment by mohanelamurugan** [comment]
> Exactly, great catch. Mohan will update.
> […](#)

@michaelcasey316 Can you check now pls. I have made the changes to move shell files to dist/bin/ directory

FYI
@aepod

**Comment by michaelcasey316** [comment]
This could work, as now `flush-varnish` and `flush-redis` will be in the `bin/` directory of your Magento Cloud project. However, I'm concerned about proliferation of files in that folder, and also with confusing people that aren't using Cloud Docker. Even if you're not using Cloud Docker, you'll still see `flush-redis` and `flush-varnish` in your `bin/` directory, which could be quite confusing when those fail to perform their stated actions. 

Maybe we can put these two commands inside of the shell script `dist/bin/magento-docker`. Then all the Cloud Docker commands would be in the same place.

**Comment by mohanelamurugan** [comment]
> This could work, as now `flush-varnish` and `flush-redis` will be in the `bin/` directory of your Magento Cloud project. However, I'm concerned about proliferation of files in that folder, and also with confusing people that aren't using Cloud Docker. Even if you're not using Cloud Docker, you'll still see `flush-redis` and `flush-varnish` in your `bin/` directory, which could be quite confusing when those fail to perform their stated actions.
> 
> Maybe we can put these two commands inside of the shell script `dist/bin/magento-docker`. Then all the Cloud Docker commands would be in the same place.

Understood. I have moved the files to `dist/bin/magento-docker`

**Review by michaelcasey316 — DISMISSED** [review]

**Review by bbatsche — DISMISSED** [review]

**Comment by mohanelamurugan** [comment]
@bbatsche @michaelcasey316 : A merge conflict has been resolved. Can you guys please
re review this PR ?

**Review by michaelcasey316 — APPROVED** [review]

**Review by bbatsche — APPROVED** [review]

**Comment by mveeramneni** [comment]
QA Approved

**Pull request merged**